component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -205,9 +206,11 @@ public boolean isValidAssertion(SignedJWT signedJWT, String requestUrl) throws O
                         consumerKey, OAuth2ErrorCodes.SERVER_ERROR);
             }
 
-            preventTokenReuse = !JWTServiceDataHolder.getInstance()
-                    .getPrivateKeyJWTAuthenticationConfigurationDAO()
-                    .getPrivateKeyJWTClientAuthenticationConfigurationByTenantDomain(tenantDomain).isEnableTokenReuse();
+            if (oAuthAppDO.isTokenEndpointAllowReusePvtKeyJwt() != null) {
+                preventTokenReuse = !oAuthAppDO.isTokenEndpointAllowReusePvtKeyJwt();
+            } else {
+                preventTokenReuse = DEFAULT_VALUE_FOR_PREVENT_TOKEN_REUSE;
```

Comment:
can we assign the default value with variable initialization?

Reply:
`preventTokenReuse = DEFAULT_VALUE_FOR_PREVENT_TOKEN_REUSE;`

Remove this completely as the default value is being already set previously during class instantiation. 

Fixed with 0bee0bc455d5c9d8d51e2296748355a7792c609e